Dockerfile

```
@@ -422,15 +421,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 WORKDIR /tmp/pljava-${pljava_release}
 RUN mvn -T 1C clean install -Dmaven.test.skip -DskipTests -Dmaven.javadoc.skip=true
 # Create debian package
-RUN cp pljava-packaging/target/pljava-pg${postgresql_major}.jar /tmp/
-
-FROM base as pljava
-# Download pre-built packages
-RUN apt-get update && apt-get install -y --no-install-recommends --download-only \
-    default-jdk-headless \
-    postgresql-${postgresql_major}-pljava \
-    && rm -rf /var/lib/apt/lists/*
-RUN mv /var/cache/apt/archives/*.deb /tmp/
+RUN checkinstall -D --install=no --fstrans=no --backup=no --pakdir=/tmp --pkgname=pljava --pkgversion="\${pljava_release//_/.}" --requires=libssl-dev,default-jdk --exclude /proc/ --nodoc java -jar pljava-packaging/target/pljava-pg${postgresql_major}.jar
```

```suggestion
RUN checkinstall -D --install=no --fstrans=no --backup=no --pakdir=/tmp --pkgname=pljava --pkgversion="\${pljava_release//_/.}" --requires=default-jdk --exclude /proc/ --nodoc java -jar pljava-packaging/target/pljava-pg${postgresql_major}.jar
```

Do you need libssl-dev at runtime? I thought it's usually a build dependency

---

Didn't check - assumed it was needed since it was not removed [here](https://github.com/supabase/postgres/pull/690/files#diff-f9985ee5a6ca94a95ad0bcfc670ebf7d1114bce1003675c4383068c1e3bf2eb8R41-R43).

Might've also been needed by some other extension further down the line and we didn't properly contain extension deps, so definitely worth a check if it's needed or not.